File: src/main/java/herbaccara/boot/autoconfigure/prevent/duplicate/PreventDuplicateRequestFilterProperties.java
package herbaccara.boot.autoconfigure.prevent.duplicate;

import herbaccara.prevent.duplicate.PreventDuplicateRequestFilter;
import herbaccara.prevent.duplicate.RequestUriType;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.http.HttpMethod;

import java.time.Duration;
import java.util.List;

@ConfigurationProperties(prefix = "prevent.duplicate")
public class PreventDuplicateRequestFilterProperties {

    private Duration timeout = PreventDuplicateRequestFilter.DEFAULT_TIMEOUT;
    private Integer order = PreventDuplicateRequestFilter.DEFAULT_FILTER_ORDER;
    private List<String> urlPatterns = List.of("/*");
    private List<HttpMethod> preventHttpMethods = PreventDuplicateRequestFilter.DEFAULT_PREVENT_HTTP_METHODS;
    private RequestUriType requestUriType = PreventDuplicateRequestFilter.DEFAULT_REQUEST_URI_TYPE;
    private Boolean withQueryString = PreventDuplicateRequestFilter.DEFAULT_WITH_QUERY_STRING;
    private Integer errorStatusCode = PreventDuplicateRequestFilter.DEFAULT_ERROR_STATUS_CODE.value();
    private String errorMessage = PreventDuplicateRequestFilter.DEFAULT_ERROR_MESSAGE;

    public Duration getTimeout() {
        return timeout;
    }

    public void setTimeout(final Duration timeout) {
        this.timeout = timeout;
    }

    public Integer getOrder() {
        return order;
    }

    public void setOrder(final Integer order) {
        this.order = order;
    }

    public List<String> getUrlPatterns() {
        return urlPatterns;
    }

    public void setUrlPatterns(final List<String> urlPatterns) {
        this.urlPatterns = urlPatterns;
    }

    public List<HttpMethod> getPreventHttpMethods() {
        return preventHttpMethods;
    }

    public void setPreventHttpMethods(final List<HttpMethod> preventHttpMethods) {
        this.preventHttpMethods = preventHttpMethods;
    }

    public RequestUriType getRequestUriType() {
        return requestUriType;
    }

    public void setRequestUriType(final RequestUriType requestUriType) {
        this.requestUriType = requestUriType;
    }

    public Boolean getWithQueryString() {
        return withQueryString;
    }

    public void setWithQueryString(final Boolean withQueryString) {
        this.withQueryString = withQueryString;
    }

    public Integer getErrorStatusCode() {
        return errorStatusCode;
    }

    public void setErrorStatusCode(final Integer errorStatusCode) {
        this.errorStatusCode = errorStatusCode;
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public void setErrorMessage(final String errorMessage) {
        this.errorMessage = errorMessage;
    }
}

File: src/main/kotlin/herbaccara/boot/autoconfigure/prevent/duplicate/PreventDuplicateRequestFilterProperties.kt
// package herbaccara.boot.autoconfigure.prevent.duplicate
//
// import herbaccara.prevent.duplicate.PreventDuplicateRequestFilter
// import herbaccara.prevent.duplicate.RequestUriType
// import org.springframework.boot.context.properties.ConfigurationProperties
// import org.springframework.http.HttpMethod
// import java.time.Duration
//
// @ConfigurationProperties(prefix = "prevent.duplicate")
// data class PreventDuplicateRequestFilterProperties(
//    val timeout: Duration = PreventDuplicateRequestFilter.DEFAULT_TIMEOUT,
//    val order: Int = PreventDuplicateRequestFilter.DEFAULT_FILTER_ORDER,
//    val urlPatterns: List<String> = listOf("/*"),
//    val preventHttpMethods: List<HttpMethod> = PreventDuplicateRequestFilter.DEFAULT_PREVENT_HTTP_METHODS,
//    val requestUriType: RequestUriType = PreventDuplicateRequestFilter.DEFAULT_REQUEST_URI_TYPE,
//    val withQueryString: Boolean = PreventDuplicateRequestFilter.DEFAULT_WITH_QUERY_STRING,
//    val errorStatusCode: Int = PreventDuplicateRequestFilter.DEFAULT_ERROR_STATUS_CODE.value(),
//    val errorMessage: String = PreventDuplicateRequestFilter.DEFAULT_ERROR_MESSAGE
// )
